### ISSUE TITLE
One constructor, zero member type with phantom type doesn't compile

### ISSUE THREAD

**Issue by LGLO** [Frege/frege#260]
Having algebraic data type with zero members and just one constructor like below creates java code that doesn't compile. Minimal example:
```haskell
module Phantom where
-- 1.
data MyT a = MyT where
  phantomString = MyT :: MyT String
-- 2.
--data MyT a = MyT {x :: Int} where  
--  phantomString = MyT 0 :: MyT String
-- 3.
--data MyT a = MyT | DontInstantiateMe where  
--  phantomString = MyT :: MyT String
```
All three compiles to Java code, but 1. results in code that doesn't compile:
```java
final public static class TMyT extends frege.runtime.Algebraic  {
  private TMyT() {}
  final public int _constructor() {
    return 0;
  }
  final public static TMyT mk() {
    return new TMyT();
  }
  final public static TMyT phantomString = TMyT.it;
}
```
`javac` complains it cannot find symbols `TMyT.it`.
2. and 3. are compiled by javac just fine.

**Comment by Ingo60** [comment]
Will be fixed with the 3.24 release.

**Comment by Ingo60** [comment]
Checked that this is fixed in 3.24.16.
